src/batInnerShuffle.ml

```diff
@@ -0,0 +1,12 @@
+let array_shuffle ?state a =
+  let random_int state n = match state with
+    | None -> Random.int n
+    | Some s -> Random.State.int s n in
+  for n = Array.length a - 1 downto 1 do
+    let k = random_int state (n + 1) in
+    if k <> n then
```

Comment:
I don't think it actually will always run: it would have a hard time given that it uses a variable that is local to the branch ;-) I think the code is behaving as it should, but you are correct that multi-line conditionals should be bracketed to avoid confusion.